Bank account verification plugin

tomer-shvadron opened the pull request:
- **feat(plugins): add bank account verification plugin**
- **feat(bank-account-verification): enhance bank account verification logic and structure**
- **fix(middleware): enable setting entities in middleware classes**


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

Based on the comprehensive summary, here are the updated release notes:

- **New Features**
  - Added Bank Account Verification plugin to the workflow core system.
  - Enhanced plugin system with new validation and transformation utilities.

- **Improvements**
  - Streamlined response transformation logic in the API plugin.
  - Updated handling of bank account verification and workflow logic.
  - Improved error handling and logging across multiple services.
  - Refined type definitions for plugin payload properties.
  - Adjusted logic for filtering workflows based on entity types.

- **Dependency Updates**
  - Multiple packages updated to latest minor versions.
  - Consistent version bumps across the Ballerine ecosystem.

- **Bug Fixes**
  - Minor code consistency improvements.
  - Corrected method naming conventions.

- **Chores**
  - Updated changelog and package version numbers.
  - Streamlined code structure in various services.
  - Removed unnecessary console log statements for environment variable loading.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
feat(plugins): add bank account verification plugin

- Create BankAccountVerificationPlugin with necessary payload validation
- Update PluginKind and pluginsRegistry to include the new plugin

(this code is so tightly coupled, it's basically a code marriage without a prenup)

tomer-shvadron committed:
feat(bank-account-verification): enhance bank account verification logic and structure

- Add memoization for bank account verification flag
- Refactor payload validation to use safeParse for error handling
- Improve logging by standardizing plugin name usage

(Your error handling is so convoluted, it's practically a labyrinth with no exit)

tomer-shvadron committed:
fix(middleware): enable setting entities in middleware classes

- Remove commented-out lines to set the entity object correctly
- Update token, admin, and auth key middlewares for improved functionality

(Your middleware logic is so convoluted, even Rubik's Cube would give up)

tomer-shvadron committed:
chore(release): bump package versions and update changelogs

- Increment versions for backoffice, kyb-app, headless-example, and workflows-service
- Update dependencies for workflow-browser-sdk and workflow-node-sdk

(Your version management is so exciting, it makes watching paint dry feel like a thriller)

tomer-shvadron committed:
Revert "chore(release): bump package versions and update changelogs"

This reverts commit fe9fbf699e9784cb1304c89771bcd771ba999788.

tomer-shvadron committed:
chore(workflows): update workflow configurations and dependencies

- Add node_modules and config files to .eslintignore
- Modify permissions to allow write access in GitHub workflows
- Update Dockerfile paths for various services

(Your workflow dependencies are boxing each other like they’re auditioning for a soap opera)

tomer-shvadron committed:
chore(all): update package versions and dependencies

- Bump versions across multiple packages
- Update dependencies for improved stability

(Your dependency updates look like they were done in a race against time)

alonp99 committed:
Merge branch 'dev' into bal-3173

tomer-shvadron committed:
Merge branch 'dev' into bal-3173

tomer-shvadron committed:
chore: bump versions and update dependencies across multiple packages

- Bump version for backoffice-v2 to 0.7.92 and update dependencies
- Bump version for kyb-app to 0.3.107 and update dependencies
- Bump version for several other packages and update respective dependencies

(Your dependency management is surprisingly more organized than my sock drawer)

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts:
```diff
@@ -0,0 +1,221 @@
+    payload,
+    ...pluginParams
+  }: IApiPluginParams & { payload: BankAccountVerificationPlugin['payload'] }) {
+    super({
```
There are a lot of issues with the block - as a newcomer ill be very confused in that point (im aware its probably something you saw in other places, but lets try cleanning up a bit)

Something the pops out right away is why its not a part of the parent constructor.
It seems like something weve done with transfomers to be quick but shouldve been integral  part of the plugin system

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts:
```diff
@@ -0,0 +1,221 @@
+  };
+};
+
+export class BankAccountVerificationPlugin extends ApiPlugin {
```
I remember there is also BallerineAPIPlugin, i dont remember exactly its purpose but wonder if it should be used here - please check

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts:
```diff
@@ -0,0 +1,221 @@
+    if (this.request?.transformers) {
+      requestPayload = await this.transformData(this.request.transformers, context);
+
+      const { isValidRequest, errorMessage } = await this.validateContent(
```
Why would a static plugin like that will have request/reponse schema coming from configuration?

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts:
```diff
@@ -0,0 +1,221 @@
+
+      if (!validatedPayload.success) {
+        return this.returnErrorResponse(
+          `Invalid payload: ${JSON.stringify(validatedPayload.error.errors)}`,
```
Add the plugin name

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts:
```diff
@@ -0,0 +1,221 @@
+
+      return {};
+    } catch (error) {
+      logger.error('Error occurred while sending an API request', { error });
```
add plugin name

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts:
```diff
@@ -0,0 +1,221 @@
+    } catch (error) {
+      logger.error('Error occurred while sending an API request', { error });
+
+      return this.returnErrorResponse(isErrorWithMessage(error) ? error.message : 'Unknown error');
```
add plugin name

tomer-shvadron committed:
Merge branch 'dev' into bal-3173

tomer-shvadron committed:
refactor(plugins): rename ballerine-plugin to ballerine-api-plugin

- Update import paths to reflect the plugin renaming
- Remove deprecated transformer logic in relevant plugins
- Enhance response handling and additional logging for clarity

(Your variable names might as well be in a different language, they're that unrecognizable)

tomer-shvadron committed:
fix(plugin): correct transformers order in handleJmespathTransformers

- Swap request and response transformers to ensure proper handling
- Enhance plugin's functionality for better consistency

(Your code has more twists than a mystery novel with a confused plot)

tomer-shvadron committed:
Merge branch 'dev' into bal-3173

tomer-shvadron committed:
chore: bump versions across multiple packages and update dependencies

- Bump versions for numerous packages to the latest patch
- Update dependencies to their respective latest versions

(With all these updates, even time itself might start feeling outdated)

tomer-shvadron committed:
chore(deps): update vite dependency to use @types/node@20.9.2

- Upgrade @types/node version for vite to ensure compatibility
- Adjust affected packages to use the latest type definitions

(your dependencies are like your jokes; outdated and in desperate need of an upgrade)

tomer-shvadron committed:
test(external-plugin): enhance plugin tests with additional properties

- Add invokedAt validation to plugin response objects
- Include name and error fields in error response assertions

(These tests could use more coverage than an overzealous sunscreen salesman)

tomer-shvadron committed:
Merge branch 'dev' into bal-3173
